kore/src/Kore/Step/Simplification/Pattern.hs

```diff
@@ -57,10 +58,13 @@ simplify
         )
     => Pattern variable
     -> simplifier (OrPattern variable)
-simplify pattern' = do
-    orSimplifiedTerms <- simplifyConditionalTermToOr predicate term
-    fmap OrPattern.fromPatterns . Branch.gather $ do
-        simplifiedTerm <- Branch.scatter orSimplifiedTerms
-        Pattern.mergeWithPredicate predicate simplifiedTerm
+simplify pattern' =
+    if isSimplified pattern'
```

Comment:
I think this check should go in `Kore.Step.Simplification.TermLike.simplifyInternal`. The reason is: this will re-simplify the _entire_ configuration if any part is unsimplified, but if we put the check in `simplifyInternal`, it will avoid re-simplifying _parts_ of the configuration that are unchanged.